test/tools/ossfuzz/protoToAbiV2.h

```diff
@@ -125,79 +130,40 @@ class ProtoConverter
 		PUBLIC,
 		EXTERNAL
 	};
-	enum class DataType
-	{
-		BYTES,
-		STRING,
-		VALUE,
-		ARRAY
-	};
-
-	void visit(BoolType const&);
-
-	void visit(IntegerType const&);
-
-	void visit(FixedByteType const&);
-
-	void visit(AddressType const&);
-
-	void visit(ArrayType const&);
-
-	void visit(DynamicByteArrayType const&);
-
-	void visit(StructType const&);
-
-	void visit(ValueType const&);
-
-	void visit(NonValueType const&);
-
-	void visit(Type const&);
-
-	void visit(VarDecl const&);
-
-	void visit(TestFunction const&);
-
+	std::pair<std::string, std::string> visit(VarDecl const&);
+	std::string visit(TestFunction const&, std::string const&);
 	void visit(Contract const&);
 
-	std::string getValueByBaseType(ArrayType const&);
-
-	DataType getDataTypeByBaseType(ArrayType const& _x);
+	template <typename T>
+	std::pair<std::string, std::string> processType(T const& _type, bool _isValueType);
 
-	void resizeInitArray(
-		ArrayType const& _x,
-		std::string const& _baseType,
+	template <typename T>
+	std::pair<std::string, std::string> assignChecker(
 		std::string const& _varName,
-		std::string const& _paramName
-	);
-
-	unsigned resizeDimension(
-		bool _isStatic,
-		unsigned _arrayLen,
-		std::string const& _var,
-		std::string const& _param,
-		std::string const& _type
+		std::string const& _paramName,
+		T _type
 	);
 
-	void resizeHelper(
-		ArrayType const& _x,
-		std::vector<std::string> _arrStrVec,
-		VecOfBoolUnsigned _arrInfoVec,
-		std::string const& _varName,
-		std::string const& _paramName
-	);
-
-	// Utility functions
-	void appendChecks(DataType _type, std::string const& _varName, std::string const& _rhs);
-
-	void addVarDef(std::string const& _varName, std::string const& _rhs);
-
-	void addCheckedVarDef(
-		DataType _type,
+	template <typename T>
+	std::pair<std::string, std::string> varDecl(
 		std::string const& _varName,
 		std::string const& _paramName,
-		std::string const& _rhs
+		T _type,
+		bool _isValueType,
+		std::string const& _location
 	);
+	std::pair<std::string, std::string> visit(Type const&);
```

Comment:
It might be worth creating `alias StringPair = std::pair<std::string, std::string>`?